lib/extensions/attachable/active_record/base.rb

```diff
@@ -27,6 +41,51 @@ def files=(files)
         attachment_references.build(file: file)
       end
     end
+
+    private
+
+    # Delete the attachment references which are not in the content.
+    def update_attachment_references
+      return if attachment_references.empty?
+
+      attachment_reference_ids = active_attachment_reference_ids
```

Comment:
Sorry I forgot to mention that there is another case: where a model can have attachments in content + attachments directly mounted on the model it self. 

e.g. for `Assessment`, it can have files directly, but it also can have images in its description, assign ids direct will remove files from the assessment.